gandalf.js

```diff
@@ -11,19 +11,22 @@
  */
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.